samples/gen-go-db/server/db/interface.go

```diff
@@ -145,6 +163,17 @@ type SchoolApp struct {
 	App    string
 }
 
+// GetTeacherSharingRulesByTeacherAndSchoolAppPageInput is the query input to GetTeacherSharingRulesByTeacherAndSchoolAppPage.
+type GetTeacherSharingRulesByTeacherAndSchoolAppPageInput struct {
+	// StartingAfter is a required specification of an (exclusive) starting point.
+	StartingAfter *models.TeacherSharingRule
```

Comment:
1) why is this required? I'd expect an option similar to `Scan` where I don't specify a starting point
2) I think this should be `StartingAt`, not `StartingAfter`, since it is an *inclusive* starting point. I.e. the comparison is >= or <= in the query (depending on descending or not).

Reply:
1. if a string ExpressionAttributeValues is empty Dynamo returns this error 
`ValidationException: One or more parameter values were invalid: An AttributeValue may not contain an empty string`
2. The generated method uses ExclusiveStartKey to avoid the duplicated ending and first result when paging. The comparisons are currently incorrect and were copied from existing methods and are effectively > and <, not >= and <=.

Reply:
re: 1) it looks like you can omit ExpressionAttributeNames and ExpressionAttributeValues: https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Query.html (they are both marked required: No)

re: 2) I see... I anticipate that the use case for the existing query methods will be missed, i.e. there are use cases where we want to list out objects from an index starting *at* a specific place. One way you could potentially do this is to accept both `StartingAt` and `StartingAfter`, where both are optional pointers. If `StartingAt` is specified, you populate `ExpressionAttributeNames`, `ExpressionAttributeValues`, `KeyConditionExpression`. If `StartingAfter` is specified, you populate `ExclusiveStartKey`

Reply:
nice. I'll try these things.

Reply:
I tried these things. It looks like these params are actually required. :fakenews:

No ExpressionAttributeNames and ExpressionAttributeValues
`ValidationException: Invalid KeyConditionExpression: An expression attribute name used in the document path is not defined; attribute name: #DISTRICT
	status code: 400, request id: 90e12efd-f920-4f9b-a429-ea9c5d176c63`

No ExpressionAttributeNames, ExpressionAttributeValues, or KeyConditionExpression
`ValidationException: Either the KeyConditions or KeyConditionExpression parameter must be specified in the request.
	status code: 400, request id: a1a55708-6ab6-4e5a-b195-e64a94d8ede8`

But I can still implement the StartingAt by optionally having the ExclusiveStartKey, everything else seems to be required in practice.

Reply:
You could enforce that either `StartingAt` or `StartingAfter` is specified (not both), and in the case where `StartingAfter` is specified, use those values in the `KeyConditionExpression` as if they were `StartingAt` (i.e. with the same `>=`/`<=`). It'll be irrelevant since the exclusive start key will be more constraining, but :shrug:

Reply:
I went with. That cool?

```
input: {
  StartingAt: model
  Exclusive: bool
}
```